src/cache/cache_operations.rs

```diff
@@ -0,0 +1,149 @@
+use crate::cache::cache_op_executors::{cache_response, invalidate, request_cached};
+use crate::cache::Cache;
+use crate::config::{
+    default_request_timeout, request_cache_duration, request_error_cache_duration,
+};
+use crate::utils::errors::ApiResult;
+use rocket::response::content;
+use serde::Serialize;
+
+pub enum Database {
+    Info = 1,
+    Default = 2,
+}
+
+pub struct Invalidate {
+    pattern: InvalidationPattern,
+    database: Database,
+}
+
+pub enum InvalidationPattern {
+    FlushAll,
+    RequestsResponses(String),
+}
+
+impl Invalidate {
+    pub fn new(pattern: InvalidationPattern) -> Self {
+        Invalidate {
+            pattern,
+            database: Database::Default,
+        }
+    }
+
+    pub fn pattern(&mut self, pattern: InvalidationPattern) -> &mut Self {
```

Comment:
Not sure if we really need it (but also doesn't harm)

Reply:
will kick it out as well, no point on having it :)